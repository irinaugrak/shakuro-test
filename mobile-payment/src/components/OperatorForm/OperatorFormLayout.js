import React from 'react';
import Select from 'react-select';
import ReactSVG from 'react-svg';
import PropTypes from 'prop-types';

import refresh_img from '../../assets/images/refresh.svg';

const OperatorFormLayout = ({
                              operatorsList,
                              isOperatorsListLoading,
                              isOperatorsListError,
                              selectedOperator,
                              onSelectChange,
                              onOperatorFormSubmit,
                              onOperatorFormRefreshClick
                            }) => {
  const operatorForm =
    isOperatorsListLoading ?
      <div className='spinner'/>
      :
      isOperatorsListError ?
        <div className='mobile-payment__operator-form-error'>
          <p className='mobile-payment__operator-form-error-text'>Что-то пошло не так. Пожалуйста, обновите
            страницу.</p>
          <ReactSVG
            path={refresh_img}
            className='mobile-payment__operator-form-error-refresh-img'
            style={{
              width: 50,
              height: 50
            }}
            onClick={onOperatorFormRefreshClick}
          />
        </div>
        :
        <div className='mobile-payment__operator-form'>
          <p className='mobile-payment__operator-legend'>Выберите оператора для того, чтобы перейти к форме
            оплаты</p>
          <Select
            value={selectedOperator}
            options={operatorsList}
            onChange={onSelectChange}
            noResultsText='Нет результатов'
            placeholder='Выберите оператора'
          />
          <a
            onClick={selectedOperator ? onOperatorFormSubmit : e => e.preventDefault()}
            className={
              selectedOperator ?
                'mobile-payment__operator-to-pay'
                :
                'mobile-payment__operator-to-pay mobile-payment__operator-to-pay--disabled'
            }
          >
            Перейти к оплате
          </a>
        </div>;

  return (
    <div className='mobile-payment__operator'>
      <div className='mobile-payment__operator-wrapper'>
        {operatorForm}
      </div>
    </div>
  )
};

export default OperatorFormLayout;

OperatorFormLayout.propTypes = {
  operatorsList: PropTypes.array,
  isOperatorsListLoading: PropTypes.bool.isRequired,
  isOperatorsListError: PropTypes.bool.isRequired,
  selectedOperator: PropTypes.shape({
    value: PropTypes.number,
    label: PropTypes.string
  }),
  onSelectChange: PropTypes.func.isRequired,
  onOperatorFormSubmit: PropTypes.func.isRequired,
  onOperatorFormRefreshClick: PropTypes.func.isRequired
};