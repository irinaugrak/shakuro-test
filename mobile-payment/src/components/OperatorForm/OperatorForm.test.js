import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';

import OperatorFormLayout from './OperatorFormLayout';

const func = jest.fn();

const setOperatorFormLayoutParams = params => (
  {
    operatorsList: params.operatorsList,
    selectedOperator: params.selectedOperator,
    isOperatorsListLoading: params.isOperatorsListLoading,
    isOperatorsListError: params.isOperatorsListError,
    onOperatorFormSubmit: func,
    onOperatorFormRefreshClick: func,
    onSelectChange: func
  }
);

const OperatorFormLayoutTestCaseCreate = testCase => {
  const params = setOperatorFormLayoutParams(testCase);
  const element = renderer
    .create(
      <MemoryRouter>
        <OperatorFormLayout
          {...params}
        />
      </MemoryRouter>
    )
    .toJSON();

  expect(element).toMatchSnapshot();
};

it('renders OperatorFormLayout correctly', () => {
  const testCases = [
    {
      operatorsList: null,
      selectedOperator: null,
      isOperatorsListLoading: true,
      isOperatorsListError: false
    },
    {
      operatorsList: [{value: 1, label: 'Мегафон'}],
      selectedOperator: {value: 1, label: 'Мегафон'},
      isOperatorsListLoading: false,
      isOperatorsListError: false
    },
    {
      operatorsList: [{value: 1, label: 'Мегафон'}],
      selectedOperator: null,
      isOperatorsListLoading: false,
      isOperatorsListError: false
    },
    {
      operatorsList: null,
      selectedOperator: null,
      isOperatorsListLoading: false,
      isOperatorsListError: true
    }
  ];

  testCases.map(item => {
    OperatorFormLayoutTestCaseCreate(item);
  });
});