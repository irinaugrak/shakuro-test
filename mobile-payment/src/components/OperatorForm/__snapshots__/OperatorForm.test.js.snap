// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PaymentFormLayout loading correctly 1`] = `
<div
  className="mobile-payment__operator"
>
  <div
    className="mobile-payment__operator-wrapper"
  >
    <div
      className="spinner"
    />
  </div>
</div>
`;

exports[`renders PaymentFormLayout loading correctly 2`] = `
<div
  className="mobile-payment__operator"
>
  <div
    className="mobile-payment__operator-wrapper"
  >
    <div
      className="mobile-payment__operator-form"
    >
      <p
        className="mobile-payment__operator-legend"
      >
        Выберите оператора для того, чтобы перейти к форме оплаты
      </p>
      <div
        className="Select has-value is-clearable is-searchable Select--single"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-2--value"
          >
            <div
              className="Select-value"
              style={undefined}
              title={undefined}
            >
              <span
                aria-selected="true"
                className="Select-value-label"
                id="react-select-2--value-item"
                role="option"
              >
                Мегафон
              </span>
            </div>
            <div
              className="Select-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-activedescendant="react-select-2--value"
                aria-describedby={undefined}
                aria-expanded="false"
                aria-haspopup="false"
                aria-label={undefined}
                aria-labelledby={undefined}
                aria-owns=""
                className={undefined}
                id={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5px",
                  }
                }
                tabIndex={undefined}
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
          <span
            aria-label="Clear value"
            className="Select-clear-zone"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            title="Clear value"
          >
            <span
              className="Select-clear"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&times;",
                }
              }
            />
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
      <a
        className="mobile-payment__operator-to-pay"
        onClick={[Function]}
      >
        Перейти к оплате
      </a>
    </div>
  </div>
</div>
`;

exports[`renders PaymentFormLayout loading correctly 3`] = `
<div
  className="mobile-payment__operator"
>
  <div
    className="mobile-payment__operator-wrapper"
  >
    <div
      className="mobile-payment__operator-form"
    >
      <p
        className="mobile-payment__operator-legend"
      >
        Выберите оператора для того, чтобы перейти к форме оплаты
      </p>
      <div
        className="Select is-clearable is-searchable Select--single"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-3--value"
          >
            <div
              className="Select-placeholder"
            >
              Выберите оператора
            </div>
            <div
              className="Select-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-activedescendant="react-select-3--value"
                aria-describedby={undefined}
                aria-expanded="false"
                aria-haspopup="false"
                aria-label={undefined}
                aria-labelledby={undefined}
                aria-owns=""
                className={undefined}
                id={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5px",
                  }
                }
                tabIndex={undefined}
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
      <a
        className="mobile-payment__operator-to-pay mobile-payment__operator-to-pay--disabled"
        onClick={[Function]}
      >
        Перейти к оплате
      </a>
    </div>
  </div>
</div>
`;

exports[`renders PaymentFormLayout loading correctly 4`] = `
<div
  className="mobile-payment__operator"
>
  <div
    className="mobile-payment__operator-wrapper"
  >
    <div
      className="mobile-payment__operator-form-error"
    >
      <p
        className="mobile-payment__operator-form-error-text"
      >
        Что-то пошло не так. Пожалуйста, обновите страницу.
      </p>
      <div
        className="mobile-payment__operator-form-error-refresh-img"
        onClick={[Function]}
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
    </div>
  </div>
</div>
`;
